Lesson10_Task01.py

```diff
@@ -0,0 +1,26 @@
+"""
+Lesson10 Task01.
+
+open input.txt file
+and find all small english letters that match such conditions:
+target small letter round exactly with three capital english letters
+on the left and on the right
+Match examples:
+sdTRYaUVKn  -> matches "a"
+NTRSjARTb   -> no match ( not exactly 3 capital letters on the left)
+zDFGbOPNq   -> matches "b"
+Print Output as : "Result: "<your_found_human_readable_string">
+"""
+
+import re
+
+try:
+    with open('input.txt', 'r') as input_txt:
+        data = input_txt.read()
+        let = re.findall('(?<=[^A-Z])[A-Z]{3}[a-z][A-Z]{3}(?=[a-z])', data)
```

Comment:
Very close , but not fully correct regexp. You see just try to add one line `<!--NCBaYUTbnVBNbKOL` into some input2.txt and test vs it.  'b' letter won't be captured.
If you have clear pre/post conditions and descriptions - better follow them strictly.
Here is example of possible regex
`r'(?<=[^A-Z][A-Z]{3})[a-z](?=[A-Z]{3}[^A-Z])'`
The best option of course compiled regexp with VERBOSE flag and comments inside

Reply:
fixed